File: src/main/java/br/com/leonardoferreira/jirareport/config/AuditorAwareImpl.java
package br.com.leonardoferreira.jirareport.config;

import br.com.leonardoferreira.jirareport.domain.vo.Account;
import java.util.Optional;
import org.springframework.data.domain.AuditorAware;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;

@Component
public class AuditorAwareImpl implements AuditorAware<String> {

    @Override
    public Optional<String> getCurrentAuditor() {
        return Optional
                .ofNullable(SecurityContextHolder.getContext().getAuthentication())
                .map(it -> (Account) it.getPrincipal())
                .map(Account::getUsername);
    }

}

File: application/src/main/kotlin/br/com/jiratorio/AuditorAwareImpl.kt
package br.com.jiratorio

import br.com.jiratorio.extension.account
import org.springframework.data.domain.AuditorAware
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component
import java.util.Optional

@Component
class AuditorAwareImpl : AuditorAware<String> {

    override fun getCurrentAuditor(): Optional<String> =
        Optional.ofNullable(SecurityContextHolder.getContext().account?.username)

}
